ee/desktop/runner/server/server.go

```diff
@@ -0,0 +1,144 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/kolide/kit/ulid"
+)
+
+type RunnerServer struct {
+	server                          *http.Server
+	listener                        net.Listener
+	logger                          log.Logger
+	desktopProcAuthTokens           map[string]string
+	mutex                           sync.Mutex
+	controlRequestIntervalOverrider controlRequestIntervalOverrider
+}
+
+const (
+	HealthCheckEndpoint = "/health"
+	MenuOpenedEndpoint  = "/menuopened"
+)
+
+type controlRequestIntervalOverrider interface {
+	SetControlRequestIntervalOverride(time.Duration, time.Duration)
+}
+
+func New(logger log.Logger, controlRequestIntervalOverrider controlRequestIntervalOverrider) (*RunnerServer, error) {
+	listener, err := net.Listen("tcp", "localhost:0")
+	if err != nil {
+		return nil, err
+	}
+
+	rs := &RunnerServer{
+		listener:                        listener,
+		logger:                          logger,
+		desktopProcAuthTokens:           make(map[string]string),
+		controlRequestIntervalOverrider: controlRequestIntervalOverrider,
+	}
+
+	if rs.logger != nil {
+		rs.logger = log.With(rs.logger, "component", "desktop_runner_root_server")
+	}
+
+	mux := http.NewServeMux()
+
+	// health check endpoint
+	mux.HandleFunc(HealthCheckEndpoint, func(w http.ResponseWriter, r *http.Request) {
+		if r.Body != nil {
+			r.Body.Close()
+		}
+	})
+
+	// menu opened endpoint
+	mux.HandleFunc(MenuOpenedEndpoint, func(w http.ResponseWriter, r *http.Request) {
+		if r.Body != nil {
+			r.Body.Close()
+		}
+
+		controlRequestIntervalOverrider.SetControlRequestIntervalOverride(5*time.Second, 1*time.Minute)
```

Comment:
Could we make these durations consts so it's easier to identify/update them in the future?

Reply:
done